src/cogitate.py

```diff
@@ -1,58 +1,230 @@
+"""Command Line Interface for the Cogitate tool."""
 
-# from data_collection import collect_commits
 import argparse
-
-# from pprint import pprint
-
-# from driller import find_repositories
-
+import os
+import validators
 import data_collection
+import data_processor
 import json_handler
 
 
-def main():
+def main(args):
     """Execute the CLI."""
-    args = retrieve_arguments()
-
-    # Currently only validates the PyGithub repository
+    if args["testwithprintargs"] == "y":
+        for key, value in args.items():
+            print(key, ":", value)
+        return
+    else:
+        pass
+    # Assess PyGithub access through token and repo path
     repository = data_collection.authenticate_repository(args["token"], args["repo"])
-
-    # Temporary structure given issue retrieval is the only function
-    contributor_data = data_collection.initialize_contributor_data(
-        "contributor_data_template"
-    )
-    contributor_data = data_collection.retrieve_issue_data(
-        repository, args["state"], contributor_data
-    )
-
-    # Intermediate between data_collection and data_processor
-    json_handler.write_dict_to_json_file(contributor_data, "contributor_data")
+    # Assess PyDriller access with link validator method
+    if repository == "INVALID" or link_validator(args["link"]) is False:
+        print("Cannot authenticate repository.")
+        return
+    else:
+        # Populate json file
+        data_collection.collect_and_add_raw_data_to_json(
+            args["link"], "raw_data_storage"
+        )
+        # calculate metrics to be used for team evaluation
+        issue_dict = {}
+        issue_dict = data_collection.retrieve_issue_data(
+            repository, args["state"], issue_dict
+        )
+        individual_metrics_dict = data_collection.calculate_individual_metrics()
+        merged_dict = data_collection.merge_metric_and_issue_dicts(
+            individual_metrics_dict, issue_dict
+        )
+        updated_dict = data_processor.add_new_metrics(merged_dict)
+        # write dictionary to the json file.
+        json_handler.write_dict_to_json_file(
+            updated_dict, "individual_metrics_storage.json"
+        )
+        # merge duplicate usernames if user requests to
+        if args["runmerge"]:
+            while True:
+                data_collection.print_individual_in_table(
+                    data_dict=updated_dict, headings=[],
+                )
+                name_to_keep = input("Please enter the username to keep")
+                name_to_merge = input("Please enter the username to merge")
+                updated_dict = data_collection.merge_duplicate_usernames(
+                    updated_dict, name_to_keep, name_to_merge
+                )
+                cont = input("Merge another username? (y/n)")
+                if cont.lower() == "y":
+                    pass
+                else:
+                    print("Ending Username merge...")
+                    break
```

Comment:
@JMilamber do you think we should as the user if they want to merge to begin with or is that something they can decide through the commands they've used

Reply:
@noorbuchi The user specifies whether to merge usernames or not as one of the required command line arguments to run the program.